numpy/core/src/multiarray/dtype_traversal.c

```diff
@@ -454,10 +454,21 @@ npy_get_clear_void_and_legacy_user_dtype_loop(
         }
         return 0;
     }
+    else if (dtype->type_num == NPY_VOID) {
+        /* 
+         * Void dtypes can have "ghosts" of objects marking the dtype because
```

Comment:
Should we have a separate issue to remind us to fix this?

Reply:
One point is, that so long we allow you to view a structured array as a bytes/unstructured void, this has to stay in place.

Reply:
Yes, I guess that makes sense - actually quite a mess if there is an object underneath. But, for another time!